packages/hoppscotch-common/src/services/spotlight/searchers/environment.searcher.ts

```diff
@@ -289,6 +295,45 @@ export class SwitchEnvSpotlightSearcherService
     }
   )[0]
 
+  fetchTeamEnvironmentList(
+    teamID: string,
+    fn: (results: TeamEnvironment[]) => void
```

Comment:
Instead of supplying the function operating on the results as a callback, can we return the populated results from the GQL query response and operate with it?